src/rustdoc-cache.ts

```diff
@@ -0,0 +1,92 @@
+import os = require("os");
+import hashFiles = require("hash-files");
+
+import * as exec from "@actions/exec";
+import * as path from "path";
+import * as cache from "@actions/cache";
+import * as core from "@actions/core";
+import * as rustCore from "@actions-rs/core";
+
+export class RustdocCache {
+    private readonly cachePath;
+    private readonly cargo;
+    private __cacheKey = "";
+
+    constructor(cargo: rustCore.Cargo) {
+        this.cachePath = path.join("target", "semver-checks", "cache");
+        this.cargo = cargo;
+    }
+
+    async save(): Promise<void> {
+        core.info("Saving rustdoc cache...");
+        await cache.saveCache([this.cachePath], await this.cacheKey());
+    }
+
+    async restore(): Promise<boolean> {
+        core.info("Restoring rustdoc cache...");
+        core.info(`Rustdoc cache path: ${this.cachePath}.`);
+        core.info(`Rustdoc cache key: ${await this.cacheKey()}.`);
+
+        const key = await cache.restoreCache([this.cachePath], await this.cacheKey());
+        if (key) {
+            core.info(`Restored rustdoc cache.`);
+            return true;
+        } else {
+            core.info(`Rustdoc cache not found.`);
+            return false;
+        }
+    }
+
+    private async cacheKey(): Promise<string> {
+        if (!this.__cacheKey) {
+            this.__cacheKey = [
+                rustCore.input.getInput("cache-key"),
+                os.platform() as string,
+                await this.getRustcVersion(),
+                await this.getCargoSemverChecksVersion(),
+                this.getCargoLocksHash(),
+                "semver-checks-rustdoc",
+            ].join("-");
+        }
+
+        return this.__cacheKey;
+    }
+
+    private getCargoLocksHash(): string {
+        const manifestPath = rustCore.input.getInput("manifest-path") || "./";
```

Comment:
This is again a bug waiting to happen. "I'm running in the directory I'm supposed to check" is a reasonable assumption for the _action as a whole_ to make, but not for the _cache component_.

Please avoid the default and pass an explicit parameter in the constructor for this.

Reply:
I changed it similarly as above